dvc/output/base.py

```diff
@@ -288,13 +289,19 @@ def save(self):
 
         self.hash_info = self.get_hash()
 
-    def commit(self):
+    def commit(self, filter_info=None):
         if not self.exists:
             raise self.DoesNotExistError(self)
 
-        assert self.hash_info
+        hash_info = (
+            self.get_hash(filter_info) if filter_info else self.hash_info
```

Comment:
We need to filter the hash_info (done in memory from existing self.hash_info.dir_info) and not re-compute the hash (actually goes to the filesystem and computes the hash from scratch). So seems like you don't really need to change `get_hash()` method at all.

Reply:
Regarding changing get_hash method I remove this change but I am not sure how I could filter hash_info with information about filter_info in the moment of calling `dvc/output/base.py commit` method since self.hash_info.dir_info is None in this moment.